File: LeetCodeChallenges/CodeWithClass/Day04/JAVA/anotherSoln.java
package Day04.JAVA;

class anotherSoln {
    public int minOperations(String[] logs) {
        int x = 0;

        for(String s : logs)
            x += add(s, x);

        return x;
    }

    public int add(String s, int x){
        return s.charAt(1) == '.' ? x == 0 ? 0 : - 1 : s.charAt(0) == '.' ? 0 : 1;
    }
}

File: LeetCodeChallenges/CodeWithClass/Day04/python/anotherSoln.py
from typing import List

class Solution:
    def minOperations(self, logs: List[str]) -> int:
        stack = []

        for l in logs:
            if l == './':
                continue
            elif l == '../':
                if len(stack) > 0:
                    stack.pop()
            else:
                stack.append(l)
        return len(stack)